### PR TITLE
task done

### PR DESCRIPTION
**Pull request by Rommelua**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (andromash) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: for the non-read operation, we should use try-finally block checklist says

**Review by andromash — CHANGES_REQUESTED** [review]
Good job!

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java (hunk 1 above)
Comment (Rommelua) [reply src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: fixed

File: src/main/java/com/dev/cinema/exception/DataProcessingException.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.dev.cinema.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String message, Exception e) {
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/exception/DataProcessingException.java]: something is missing

Comment (Rommelua) [reply src/main/java/com/dev/cinema/exception/DataProcessingException.java]: fixed)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**